Bearded.Utilities/Noise/NoiseUtils.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+
+namespace Bearded.Utilities.Noise
+{
+    static class NoiseUtils
+    {
+        internal static T[,] GenerateRandomArray<T>(int width, int height, Func<Random, T> generator, int? seed)
```

Comment:
Instead of injecting a seed, I think it would be better to inject a Random instance. That would allow better control over where numbers come from and can prevent unnecessary allocation.

Reply:
I am happy to be convinced otherwise, but let me explain my reasoning:

One of the properties is that this method returns the exact same result if you give it the same input, i.e. it's deterministic (unless you choose not to pass in a seed). When you replace the seed with a `Random` instance, this because much more nuanced. Sure, if you pass in the "same" `Random`, it is still deterministic. However, there isn't an `Equals` method on `Random`, so the whole concept of having the "same" `Random` is a bit weird. Using a seed makes this much more obviously correct.

Like I said, happy to be convinced otherwise.